app/main.py

```diff
@@ -1 +1,31 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name, health=100):
+        self.name = name
+        self.health = health
+        self.hidden = False
+
+        Animal.alive.append(self)
+
+    def __repr__(self):
+        return f"{{Name: {self.name}," \
+               f" Health: {self.health}," \
+               f" Hidden: {self.hidden}}}"
+
+
+class Herbivore(Animal):
+
+    def hide(self):
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    @staticmethod
+    def bite(herbivore: Herbivore):
+        if herbivore.hidden is False and isinstance(herbivore, Herbivore):
```

Comment:
Change order of checks so it won`t generate AttributeError with non Herbivore object due to lazy calculation